src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/BindToInputBindingProvider.cs

```diff
@@ -50,6 +53,63 @@ public Task<IBinding> TryCreateAsync(BindingProviderContext context)
             return Task.FromResult<IBinding>(binding);
         }
 
+        public IEnumerable<Rule> GetRules()
+        {
+            var cm = (ConverterManager)_converterManager;
+            var types = cm.GetConvertersFromType(typeof(TAttribute), typeof(TType));
+                        
+            yield return new Rule
+            {
+                SourceAttribute = typeof(TAttribute),
+                UserType = new ConverterManager.ExactMatch(typeof(TType))
+            };
+
+            var converters = new Type[] { typeof(TType) };
+
+            foreach (var type in types)
+            {
+                yield return new Rule
+                {
+                    SourceAttribute = typeof(TAttribute),
+                    Converters = converters,
+                    UserType = type
+                };
+            }
+        }
+
+        // DefaultTypes (in precedence order) that we check for input bindings. 
+        private static readonly Type[] DefaultTypes = new Type[] 
+        {
```

Comment:
It looked like the ordering is different in AsyncCollector (it goes byte[], string, JObject). Should they be consistent?